algorithms/gr-dij/javascript/dijkstra_algorithm.js

```diff
@@ -0,0 +1,52 @@
+/*
+* Dijkstra's single source shortest path algorithm in JavaScript.
+ */
+ const dijkstraAlgorithm = function (graph, start) {
+
+    //the distances from the start node to all other nodes
+    var distance = [];
+    //Initial distance is "Infinity"
+    for (var i = 0; i < graph.length; i++) distance[i] = Number.MAX_VALUE;
+    //The distance from the start node to itself is 0
+    distance[start] = 0;
+
+    //it contains whether a node was already visited
+    var visited = [];
+
+    //While there are nodes left to visit
+    while (true) {
+        //find the node with the currently shortest distance from the start node
+        var shortestDistance = Number.MAX_VALUE;
+        var shortestIndex = -1;
+        for (var i = 0; i < graph.length; i++) {
```

Comment:
Codacy found an issue: ['i' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6358206)